File: src/main/java/io/github/eone666/telegramnotifier/events/PlayerQuit.java
package io.github.eone666.telegramnotifier.events;

import io.github.eone666.telegramnotifier.TelegramNotifier;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerQuitEvent;

public class PlayerQuit implements Listener {

    private final TelegramNotifier plugin;
    public PlayerQuit(TelegramNotifier pluginInstance){
        plugin = pluginInstance;
    }
    @EventHandler
    public void onPlayerQuit(PlayerQuitEvent event) {
        plugin.notifications.quit(event.getPlayer());
        plugin.playersList.quit(event.getPlayer());
    }
}

File: src/main/kotlin/io/github/eone666/telegramnotifier/events/PlayerQuit.kt
package io.github.eone666.telegramnotifier.events

import io.github.eone666.telegramnotifier.TelegramNotifier
import io.github.eone666.telegramnotifier.features.notifications.NotificationTypes
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerQuitEvent

class PlayerQuit(private val plugin: TelegramNotifier) : Listener {
    @EventHandler
    fun onPlayerQuit(event: PlayerQuitEvent) {
        plugin.notifications.send(NotificationTypes.QUIT, event.player)
        plugin.playersList.remove(event.player)
    }
}